test/Interop/Cxx/access/Inputs/non-public.h

```diff
@@ -0,0 +1,32 @@
+#ifndef NON_PUBLIC_H
+#define NON_PUBLIC_H
+
+// Override this to test structs
+#ifndef TEST_CLASS
+#define TEST_CLASS class
+#endif
+
+// Override this to test protected
+#ifndef TEST_PRIVATE
+#define TEST_PRIVATE private
+#endif
```

Comment:
I think I would prefer having a few (mostly) copy-pasted types over these macros, since the macros make it harder to debug test failures and make small incremental changes to the tests.

Reply:
I agree with the sentiment you express and inlined some of the other macros I was using.

However, I actually ended up keeping these specific overrides because the visibility of members here should actually behave exactly the same whether the record is a `struct` or a ` class`, and whether the members are `private` or `protected`. In other words, these macros here only superficially quadruple our test coverage, because they are exercising the exact same code path in ClangImporter.